src/utils/fetchInterestTags.ts

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from "react";
+import { fetchHistoryItems } from "./fetchHistoryItems";
+import { loadInterestData, clearInterestData } from "./dataUtils";
+import { useGeminiNanoResponse } from "./fetchGeminiResponse";
+import { HistoryItem } from "../types/historyItemType";
+import { Message, Sender } from "../types/messageType";
+
+export const useFetchInterestTags = () => {
+  const [interestTags, setInterestTags] = useState<string | null>(null);
+  const [historyItems, setHistoryItems] = useState<HistoryItem[]>([]);
+  const [summaryText, setSummaryText] = useState<string>("");
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [hasLocalData, setHasLocalData] = useState<boolean>(false);
+
+  const {
+    fetchGeminiNanoResponse,
+    loading,
+    messages: fetchedMessages,
+  } = useGeminiNanoResponse();
+
+  useEffect(() => {
+    loadInterestData((savedTags) => {
+      if (savedTags) {
+        setInterestTags(savedTags);
+        setMessages([{ sender: Sender.AI, text: savedTags }]);
+        setHasLocalData(true);
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!hasLocalData && !summaryText) {
+      getHistoryData();
+    }
+  }, [hasLocalData, summaryText]);
+
+  const getHistoryData = async () => {
+    const { historyItems, summaryText } = await fetchHistoryItems();
+    setHistoryItems(historyItems);
+    setSummaryText(summaryText);
+  };
+
+  useEffect(() => {
+    if (
+      summaryText &&
+      summaryText !== "No recent history items found." &&
```

Comment:
nit: Better to keep error messages in an enum as well to avoid possible typos in such logical comparisons etc., it can be changed later on